src/structures/GuildEmoji.js

```diff
@@ -162,15 +162,15 @@ class GuildEmoji extends Emoji {
         other.name === this.name &&
         other.managed === this.managed &&
         other.requiresColons === this.requiresColons &&
-        other.roles.size === this.roles.size &&
-        other.roles.every(role => this.roles.has(role.id))
+        other.roles.cache.size === this.roles.cache.size &&
+        other.roles.cache.every(role => this.roles.cache.has(role.id))
       );
     } else {
       return (
         other.id === this.id &&
         other.name === this.name &&
-        other.roles.length === this.roles.size &&
-        other.roles.every(role => this.roles.has(role))
+        other.roles.cache.length === this.roles.cache.size &&
```

Comment:
Those are "raw" api object, see [here](https://github.com/discordjs/discord.js/blob/master/src/client/actions/GuildEmojisUpdate.js#L17).
```suggestion
        other.roles.length === this.roles.cache.size &&
```